teamscale_client/client.py

```diff
@@ -108,157 +142,205 @@ def put(self, url, json=None, parameters=None, data=None):
+    def add_findings_group(self, name: str, mapping_pattern: str) -> requests.Response:
         """Adds group of findings.
 
         Args:
-            name (str): Name of group.
-            mapping_pattern (str): Regular expression to match a finding's ``typeid`` in order to belong to this group.
+            name: Name of group.
+            mapping_pattern: Regular expression to match a finding's ``typeid`` in order to belong to this group.
         Returns:
             requests.Response: request's response
         """
-        url = "%s/%s" % (self.get_global_service_url('external-findings-group'), name)
-        return self.put(url, {'groupName': name, 'mapping': mapping_pattern})
+        return self.post(
+            f"{self._api_url_version}/external-findings/groups",
+            json={'groupName': name, 'mapping': mapping_pattern}
+        )
 
-    def add_finding_descriptions(self, descriptions):
+    def add_finding_descriptions(self, descriptions: List[FindingDescription]) -> requests.Response:
         """Adds descriptions of findings.
 
         Args:
-            descriptions (list): List of :class:`FindingDescription` to add to Teamscale.
+            descriptions: List of :class:`FindingDescription` to add to Teamscale.
         Returns:
             requests.Response: request's response
         """
-        base_url = self.get_global_service_url('external-findings-description')
         response = None
         for finding_description in descriptions:
-            some_description = dict()
-            some_description['typeId'] = finding_description.typeid
-            some_description['description'] = finding_description.description
-            some_description['enablement'] = finding_description.enablement
-            some_description['name'] = finding_description.name
-            url = "%s/%s" % (base_url, finding_description.typeid)
-            response = self.put(url, some_description)
-            if response.text != 'success':
-                return response
-
+            response = self.post(
+                f"{self._api_url_version}/external-findings/descriptions",
+                data=to_json(finding_description),
+                headers={'Accept': 'application/json', 'Content-Type': 'application/json'}
+            )
         return response
 
     def update_findings_schema(self):
-        """Triggers refresh of finding groups in analysis profiles."""
-        url = self.get_global_service_url('update-findings-schema')
-        return self.get(url, {'projects': self.project})
+        """Triggers refresh of finding groups in analysis profiles.
 
-    def upload_findings(self, findings, timestamp, message, partition):
+        Warnings:
+            Uses Teamscale's internal API and not the versioned API!
+        """
+        return self.post(
+            f"{self._api_url}/projects/{self.project}/metric-update",
+            params={"projects": self.project},
```

Comment:
Is this supposed to be a query parameter? AFAICS the service does not take any parameters

Reply:
A forgotten remnant